monkey/monkey_island/cc/models/utils/document_encryptor.py

```diff
@@ -1,47 +1,45 @@
 from dataclasses import dataclass
-from typing import Callable, Type
+from typing import Callable, List, Type
 
 import dpath.util
 
 from monkey_island.cc.models.utils.field_encryptors.i_field_encryptor import IFieldEncryptor
-from monkey_island.cc.models.utils.field_encryptors.string_list_encryptor import StringListEncryptor
+
+
+class FieldNotFoundError(Exception):
+    pass
 
 
 @dataclass
 class SensitiveField:
     path: str
     path_separator = "."
-    field_type: Type[IFieldEncryptor]
-
-
-sensitive_fields = [
-    SensitiveField(path="overview.config_passwords", field_type=StringListEncryptor)
-]
+    field_encryptor: Type[IFieldEncryptor]
 
 
-def encrypt(report: dict) -> dict:
+def encrypt(sensitive_fields: List[SensitiveField], document_dict: dict) -> dict:
     for sensitive_field in sensitive_fields:
-        _apply_operation_to_report_field(
-            report, sensitive_field, sensitive_field.field_type.encrypt
+        _apply_operation_to_document_field(
+            document_dict, sensitive_field, sensitive_field.field_encryptor.encrypt
         )
 
-    return report
+    return document_dict
 
 
-def decrypt(report: dict) -> dict:
+def decrypt(sensitive_fields: List[SensitiveField], document_dict: dict) -> dict:
     for sensitive_field in sensitive_fields:
-        _apply_operation_to_report_field(
-            report, sensitive_field, sensitive_field.field_type.decrypt
+        _apply_operation_to_document_field(
+            document_dict, sensitive_field, sensitive_field.field_encryptor.decrypt
         )
-    return report
+    return document_dict
 
 
-def _apply_operation_to_report_field(
+def _apply_operation_to_document_field(
     report: dict, sensitive_field: SensitiveField, operation: Callable
 ):
     field_value = dpath.util.get(report, sensitive_field.path, sensitive_field.path_separator, None)
     if field_value is None:
-        raise Exception(
+        raise FieldNotFoundError(
```

Comment:
I agree, this is needlessly tied to MongoDB jargon due to anticipation of model-related encryption that never happened.

Reply:
I renamed it to dict_encryptor because it accepts dicts. For dicts, field is a correct term, even better than key, because these functions neither accept nor search for key. So this error should be either `KeyPathError` or stay `FieldNotFoundError`. One more strongly suggests that a typo was made when specifying path, another that the field location changed/it doesn't exist. Since this is a minor difference, I kept the name